flit/init.py

```diff
@@ -0,0 +1,178 @@
+import configparser
+import json
+import os
+from pathlib import Path
+import re
+import sys
+
+def get_data_dir():
+    """Get the directory path for flit user data files.
+    """
+    home = os.path.realpath(os.path.expanduser('~'))
+
+    if sys.platform == 'darwin':
+        d = Path(home, 'Library')
+    elif os.name == 'nt':
+        appdata = os.environ.get('APPDATA', None)
+        if appdata:
+            d = Path(appdata)
+        else:
+            d = Path(home, 'AppData', 'Roaming')
+    else:
+        # Linux, non-OS X Unix, AIX, etc.
+        xdg = os.environ.get("XDG_DATA_HOME", None)
+        d = Path(xdg) if xdg else Path(home, '.local/share')
+
+    return d / 'flit'
+
+def get_defaults():
+    try:
+        with (get_data_dir() / 'init_defaults.json').open() as f:
+            return json.load(f)
+    except FileNotFoundError:
+        return {}
+
+def store_defaults(d):
+    data_dir = get_data_dir()
+    try:
+        data_dir.mkdir(parents=True)
+    except FileExistsError:
+        pass
+    with (data_dir / 'init_defaults.json').open('w') as f:
+        json.dump(d, f, indent=2)
+
+license_names_to_classifiers = {
+    'BSD': 'License :: OSI Approved :: BSD License',
+    'GPL': 'License :: OSI Approved :: GPL License',
+    'Apache': 'License :: OSI Approved :: Apache License'
+}
+
+class IniterBase:
+    def __init__(self, directory='.'):
+        self.directory = Path(directory)
+        self.defaults = get_defaults()
+
+    def validate_email(self, s):
+        # Properly validating an email address is much more complex
+        return bool(re.match(r'.+@.+', s))
+
+    def guess_module_name(self):
+        packages, modules = [], []
+        for p in self.directory.iterdir():
+            if p.is_dir() and (p / '__init__.py').is_file():
+                if p.name not in {'test', 'tests'}:
+                    packages.append(p.name)
+            elif p.is_file() and p.suffix == '.py':
+                if p.stem not in {'setup'} and not p.name.startswith('test_'):
+                    modules.append(p.stem)
+        if len(packages) == 1:
+            return packages[0]
+        elif len(packages) == 0 and len(modules) == 1:
+            return modules[0]
+        else:
+            return None
+    
+    def update_defaults(self, author, author_email, module, home_page, license):
+        new_defaults = {'author': author, 'author_email': author_email,
+                        'license': license}
+        name_chunk_pat = r'\b{}\b'.format(re.escape(module))
+        if re.search(name_chunk_pat, home_page):
+            new_defaults['home_page_template'] = \
+                re.sub(name_chunk_pat, '{modulename}', home_page, flags=re.I)
+        if any(new_defaults[k] != self.defaults.get(k) for k in new_defaults):
+            self.defaults.update(new_defaults)
+            store_defaults(self.defaults)
+
+
+class TerminalIniter(IniterBase):
+    def prompt_text(self, prompt, default, validator):
```

Comment:
This and next one could be class methods. Or function. Depends on wether you are considering reusing them.